File: C_Programming/Day_4/program17.c
// Descriptikn : Write a program that accepts number from the user & returns 
//               its sum of factors & non-factors

#include<stdio.h>
#include<stdbool.h>

void Summation(int iNo)
{
    int iCnt = 0;
    int iSum1 = 0;
    int iSum2 = 0;

    for(iCnt = 1 ; iCnt <= iNo ; iCnt++)
    {
        if(iNo % iCnt == 0)
        {
            iSum1 = iSum1 + iCnt;
        }
        else
        {
            iSum2 = iSum2 + iCnt;
        }
    }

    printf("Sum of factors : %d\n",iSum1);
    printf("Sum of Non factors is : %d\n",iSum2);

}

int main()
{
    int iValue = 0;

    printf("Enter the number : ");
    scanf("%d",&iValue);

    Summation(iValue);

    return 0;
}

File: Python_Programming/Day_4/program17.py
# Description : Write a program that accepts number from the user & returns 
#               its sum of factors & non-factors

def SumFactorsNonFactors(No):
    Sum1 = 0
    Sum2 = 0

    for i in range(1 , No+1 , 1):
        if(No % i == 0):
            Sum1 = Sum1 + i
        else:
            Sum2 = Sum2 + i
        
    print("Summation of factors is : ",Sum1)
    print("Summation of non-factors is : ",Sum2)


def main():

    Value = 0

    print("Enter the number : ")
    Value = int(input())

    SumFactorsNonFactors(Value)

if __name__ == "__main__":
    main()